### PR TITLE
51300 devops qa ft integration

### PR DESCRIPTION
**Pull request by surendrabehara2022**
Hi,
This PR consists of:
Adding FT in QAdeploy stage.

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs [hunk 1]
```diff
@@ -41,38 +42,31 @@ public async Task Setup()
         }
 
         [Test]
-        public async Task WhenICallTheExchangeSetApiWithValidProductIdentifiers_ThenValidrequestedProductCountAndexchangeSetCellCountIsReturned()
+        public async Task WhenICallTheExchangeSetApiWithValidProductIdentifiers_ThenANonZerorequestedProductCountAndexchangeSetCellCountIsReturned()
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs:45]: Please capitalise the first letter of each word (Requested, Exchange)

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs [hunk 2]
```diff
@@ -41,38 +42,31 @@ public async Task Setup()
...
-            Assert.That((int)prodId_message.exchangeSetCellCount, Is.EqualTo(3155), $"Incorrect Exchange Set Cell Count is returned {prodId_message.exchangeSetCellCount}, instead of the expected status 3155.");
+            var apiResponse = await getproductIdentifier.GetProductIdentifiersDataAsync(ESSAuth.EssApiUrl, productIdentifiers, EssJwtToken);
+            Assert.That((int)apiResponse.StatusCode, Is.EqualTo(200), $"Incorrect status code is returned {unpResponse.StatusCode}, instead of the expected status 200.");
+
+            //verify model structure
+            await apiResponse.CheckModelStructureForSuccessResponse();
         }
 
         [Test]
         public async Task WhenICallTheExchangeSetApiWithInValidProductIdentifiers_ThenValidrequestedProductCountAndlessexchangeSetCellCountIsReturned()
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs:55]: Please capitalise the first letter of each word (Requested, Less, Exchange)

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+            //Check ExchangeSetFileUri is Not null and it is a valid Uri
+            Assert.IsNotNull(apiResponseData.Links.ExchangeSetFileUri.Href, "Response body returns null instead of valid links.");
+            Assert.IsTrue(Uri.IsWellFormedUriString(apiResponseData.Links.ExchangeSetFileUri.Href, UriKind.RelativeOrAbsolute), $"Exchange set returned file URI {apiResponseData.Links.ExchangeSetFileUri.Href}, Its not valid uri");
+
+            //Check ExchangeSetUrlExpiryDateTime is not null
+            Assert.IsNotNull(apiResponseData.ExchangeSetUrlExpiryDateTime, $"Response body returns null, Instead of valid datetime {apiResponseData.ExchangeSetUrlExpiryDateTime}.");
+
+            //Check data type of RequestedProductCount and value should not be less than zero
+            Assert.IsTrue(apiResponseData.RequestedProductCount.GetType().Equals(typeof(int)), "Responsebody returns other datatype, instead of expected Int");
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs:24]: Fix spelling (Responsebody -> Response body) to be consistent with previous messages

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            Assert.IsTrue(apiResponseData.RequestedProductCount >= 0, "Response body returns RequestedProductCount less than zero, instead of expected count should not be less than zero.");
+
+            //Check data type of ExchangeSetCellCount and value should not be less than zero
+            Assert.IsTrue(apiResponseData.ExchangeSetCellCount.GetType().Equals(typeof(int)), "Responsebody returns other datatype, instead of expected Int");
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs:28]: Fix spelling (Responsebody -> Response body) to be consistent with previous messages

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            Assert.IsTrue(apiResponseData.ExchangeSetCellCount >= 0, "Response body returns ExchangeSetCellCount less than zero, instead of expected count should not be less than zero.");
+
+            //Check data type of RequestedProductsAlreadyUpToDateCount and value should not be less than zero
+            Assert.IsTrue(apiResponseData.RequestedProductsAlreadyUpToDateCount.GetType().Equals(typeof(int)), "Responsebody returns other datatype, instead of expected Int");
```
Comment (ajasonca) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs:32]: Fix spelling (Responsebody -> Response body) to be consistent with previous messages

**Review by ajasonca — CHANGES_REQUESTED** [review]
Some minor spelling issues to fix

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs (hunk 2 above)
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs]: Hi Jason,

The fix has been applied. Please have a look.

Thanks,
Mayuresh Gawde

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs (hunk 3 above)
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs]: Hi Jason,

The fix has been applied. Please have a look.

Thanks,
Mayuresh Gawde

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs (hunk 5 above)
Comment (Mayuresh-Gawde) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/ModelStructureCheck.cs]: Hi Jason,

The fix has been applied. Please have a look.

Thanks,
Mayuresh Gawde

**Review by ajasonca — APPROVED** [review]

**Pull request merged**